test/test_trixi.jl

```diff
@@ -128,3 +128,40 @@ macro test_nowarn_mod(expr)
     end
   end
 end
+
+
+"""
+    @trixi_testset "name of the testset" #= code to test #=
+
+Similar to `@testset`, but wraps the code inside a temporary module to avoid
+namespace pollution. It also `include`s this file again to provide the
+definition of `@test_trixi_include`.
+"""
+macro trixi_testset(name, expr)
+  @assert name isa String
+  # TODO: `@eval` is evil
+  # We would like to use
+  #   mod = gensym(name)
+  # to create new module names for every test set. However, this is not
+  # compatible with the dirty hack using `@eval` to get the mapping when
+  # loading structured, curvilinear meshes. Thus, we need to use a plain
+  # module name here.
+  mod = Symbol("TrixiTestModule")
+  quote
+    @eval module $mod
```

Comment:
Why not
```suggestion
  quote
    @eval module TrixiTestModule
```
?

Reply:
Originally, I wanted to keep the namespace more clean by naming the test module based on the name of the testset and `gensym` as described in the comment above. Since that doesn't work with the `@eval` hack for loading curved meshes, I needed to do something else here. Your suggestion is fine as well and simpler for now, so I used that.

Reply:
OK, sounds good!